initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildWriter.java

```diff
@@ -213,36 +212,53 @@ private void writeScm(IndentingWriter writer, MavenScm mavenScm) {
-        if (dependencies.isEmpty()) {
+        List<Dependency> items = dependencies.ids()
+                .filter(this::isNotCodeOnly)
+                .map(dependencies::get)
+                .collect(Collectors.toList());
+        if (CollectionUtils.isEmpty(items)) {
             return;
         }
         writeElement(writer, "dependencies", () -> {
-            Collection<Dependency> compiledDependencies = writeDependencies(writer, dependencies,
+            Collection<Dependency> compiledDependencies = writeDependencies(writer, items,
                     (scope) -> scope == null || scope == DependencyScope.COMPILE);
             if (!compiledDependencies.isEmpty()) {
                 writer.println();
             }
-            writeDependencies(writer, dependencies, hasScope(DependencyScope.RUNTIME));
-            writeDependencies(writer, dependencies, hasScope(DependencyScope.COMPILE_ONLY));
-            writeDependencies(writer, dependencies, hasScope(DependencyScope.ANNOTATION_PROCESSOR));
-            writeDependencies(writer, dependencies, hasScope(DependencyScope.PROVIDED_RUNTIME));
-            writeDependencies(writer, dependencies,
-                    hasScope(DependencyScope.TEST_COMPILE, DependencyScope.TEST_RUNTIME));
+            writeDependencies(writer, items, hasScope(DependencyScope.RUNTIME));
+            writeDependencies(writer, items, hasScope(DependencyScope.COMPILE_ONLY));
+            writeDependencies(writer, items, hasScope(DependencyScope.ANNOTATION_PROCESSOR));
+            writeDependencies(writer, items, hasScope(DependencyScope.PROVIDED_RUNTIME));
+            writeDependencies(writer, items, hasScope(DependencyScope.TEST_COMPILE, DependencyScope.TEST_RUNTIME));
         });
     }
 
     private Predicate<DependencyScope> hasScope(DependencyScope... validScopes) {
         return (scope) -> Arrays.asList(validScopes).contains(scope);
     }
 
-    private Collection<Dependency> writeDependencies(IndentingWriter writer, DependencyContainer dependencies,
+    private Collection<Dependency> writeDependencies(IndentingWriter writer, List<Dependency> items,
                                                      Predicate<DependencyScope> filter) {
-        Collection<Dependency> candidates = dependencies.items().filter((dep) -> filter.test(dep.getScope()))
-                .sorted(getDependencyComparator()).collect(Collectors.toList());
+        Collection<Dependency> candidates = items.stream()
+                .filter((dep) -> filter.test(dep.getScope()))
+                .sorted(getDependencyComparator())
+                .collect(Collectors.toList());
         writeCollection(writer, candidates, this::writeDependency);
         return candidates;
     }
 
+    private boolean isNotCodeOnly(String id) {
+        io.spring.initializr.metadata.Dependency dep = metadata.getDependencies().get(id);
+        if (dep == null) {
```

Comment:
Should to check the logic, I saw a same other dependencies such as spring-boot-starter or spring-boot-starter-test are considered to be CodeOnly.
![image](https://user-images.githubusercontent.com/25423978/207498316-4ad126eb-a3b6-41a0-aa38-b6a411678223.png)

Reply:
done